nix/find-haskell-paths/parser_tests.nix

```diff
@@ -32,6 +32,23 @@ let
         expected = "foo";
       };
     };
+  cabalExecutableTests =
+    let
+      eval = s:
+        let res = parser.parseCabalExecutableName s; in
+        if res.type == "success" then res.value else res;
+    in
+    {
+      testSimple = {
+        expr = eval ''
+          executable foo-exec
+            main-is: foo.hs
+          executable bar-exec
+            main-is: bar.hs
+        '';
+        expected = [ "foo-exec" "bar-exec" ];
```

Comment:
Could expand the `expr` to insert lines before/between/after these executable stanzas?